enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,647 @@
+their network to be able to provision hosts inside their network. Reasons
+include trust, control, and reproducibility. Those users will prefer to run the
+service on-premises. There are also many customers whose network environments
+have limited or no connectivity to the internet.
+
+[Hive](https://github.com/openshift/hive/) is the kubernetes-native API for
+installing OpenShift, and it is the natural place to introduce an on-premises
+capability for agent-based cluster installation.
+
+### Goals
+
+* Expose agent-based installation capabilities as a kubernetes-native API
+through [Hive](https://github.com/openshift/hive).
+* Enable using agent-based installation capabilities to add workers to clusters
+that have been installed using - or adopted by - Hive.
+* Enable automated creation and booting of the Assisted discovery ISO for bare-metal deployments.
+* Ensure the design is extensible for installing on other platforms.
+
+### Non-Goals
+
+* Remote Worker Node support requires solving specific problems that will be
+addressed in a separate proposal. For example, it requires the ability to
+ensure that the Node joins with an appropriate role such that it will not be
+schedulable for specific workloads, such as the ingress controller.
+* Solve central machine management. ("central machine management" involves
+running machine-api-providers on a hub cluster in order to manage Machines on
+the hub which represent Nodes in spoke clusters.) While this is a foundational
+element to solving central machine management, this enhancement doesn't attempt
+to address the topic in its totality.
+* Run metal3 components on a non-baremetal cluster. Some parts of this proposal
+assume that the baremetal-operator is present on the hub cluster. Today that is
+only possible when the hub cluster itself is on the bare metal platform. Future
+work may propose running baremetal-operator on non-baremetal hub clusters.
+* Applying static network configuration to hosts. This is a desired feature
+but will be addressed in a future proposal.
+
+## Proposal
+
+### User Stories
+
+Personas for multi-cluster management:
+
+**Infra Owner** Manages physical infrastructure and configures hosts to boot
+the discovery ISO that runs the Agent.
+
+**Cluster Creator** Uses running Agents to create and grow clusters.
+
+#### Run Agent via Boot It Yourself
+
+Whether you are creating a new cluster or adding nodes to an existing cluster,
+the workflow starts by booting a discovery ISO on a host so that it runs the
+Agent. Many users have their own methods for booting ISOs on hardware. The
+Assisted Service must be able to create a discovery ISO that users can take and
+boot on hardware with their own methods.
+
+*As an Infra Owner, I can make Agents available for provisioning by creating a
+discovery ISO and using my own tooling to boot it on hosts.*
+
+#### Run Agent via Automation
+
+Many users will want an integrated end-to-end experience where they describe
```

Comment:
i don't think  it's just hardware... we need to describe the hardware, but also the network configuration, credentials to access the hardware, etc

Reply:
You are correct, basically everything that is in the InstallEnv